server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/LocalShardsBalancer.java

```diff
@@ -781,12 +781,14 @@ void allocateUnassigned() {
          */
         ShardRouting[] unassignedShards = unassigned.drain();
         List<ShardRouting> allUnassignedShards = Arrays.stream(unassignedShards).collect(Collectors.toList());
```

Comment:
Is this being used elsewhere since we are not keeping the value for `allUnassignedShards` the same.

Reply:
Remove unnecessary streams from the code. Actually, major bottleneck here was `List.removeAll `call. So did not modified any other flows.